src/components/todo-list/TodoListFooter.js

```diff
@@ -1,7 +1,22 @@
 import React from "react";
+import { connect } from "react-redux";
+import { removeTodo, filterTodos } from "../../actions";
+import { filterTypes } from "../../utils";
 import "./TodoListFooter.css";
 
 class TodoListFooter extends React.PureComponent {
+  onFilterChange = e => {
+    e.preventDefault();
+    const filter = e.target.href.split("/").pop();
+    this.props.filterTodos(filter);
+  };
+
+  clearCompleted = () => {
+    this.props.items.forEach(element =>
+      element.completed ? this.props.removeTodo(element.id) : null
+    );
+  };
```

Comment:
Create separate action for clearing completed all at once instead of calling remove action in a loop